### ISSUE TITLE
Building error: -Wno-class-memaccess, using g++ 9.3.0 on Ubuntu 20.04

### ISSUE THREAD

**Issue by razcafe** [recastnavigation/recastnavigation#497]
I was able to successfully build if I went through Recastdemo.make and Detour.make and removed all the references to -Wno-class-memaccess, but I thought I should post this here in case there's a better way to build it without messing with those files, or in case this is a bug.

```
==== Building DebugUtils (debug) ====
Creating obj/Debug/DebugUtils
DebugDraw.cpp
DetourDebugDraw.cpp
RecastDebugDraw.cpp
RecastDump.cpp
Creating lib/Debug
Linking DebugUtils
==== Building Detour (debug) ====
Creating obj/Debug/Detour
DetourAlloc.cpp
DetourAssert.cpp
DetourCommon.cpp
DetourNavMesh.cpp
DetourNavMeshBuilder.cpp
DetourNavMeshQuery.cpp
DetourNode.cpp
Linking Detour
==== Building DetourCrowd (debug) ====
Creating obj/Debug/DetourCrowd
DetourCrowd.cpp
DetourLocalBoundary.cpp
DetourObstacleAvoidance.cpp
DetourPathCorridor.cpp
DetourPathQueue.cpp
DetourProximityGrid.cpp
Linking DetourCrowd
==== Building DetourTileCache (debug) ====
Creating obj/Debug/DetourTileCache
DetourTileCache.cpp
DetourTileCacheBuilder.cpp
Linking DetourTileCache
==== Building Recast (debug) ====
Creating obj/Debug/Recast
Recast.cpp
RecastAlloc.cpp
RecastArea.cpp
RecastAssert.cpp
RecastContour.cpp
RecastFilter.cpp
RecastLayers.cpp
RecastMesh.cpp
RecastMeshDetail.cpp
RecastRasterization.cpp
RecastRegion.cpp
Linking Recast
==== Building RecastDemo (debug) ====
Creating obj/Debug/RecastDemo
fastlz.c
cc1: error: command line option ‘-Wno-class-memaccess’ is valid for C++/ObjC++ but not for C [-Werror]
cc1: all warnings being treated as errors
make[1]: *** [RecastDemo.make:150: obj/Debug/RecastDemo/fastlz.o] Error 1
make: *** [Makefile:69: RecastDemo] Error 2

```

**Comment by PacaPop** [comment]
Same issue here trying to build RecastDemo on linux.

**Comment by grahamboree** [comment]
Hmm this is likely a misconfiguration or bug in premake or our premake setup, assuming you're using that and not cmake.  If you can figure out a fix that doesn't require manually updating the generated make files, we'd greatly appreciate it.  Otherwise I can take a look later this week.  Thanks for reporting this!

**Comment by SgtVincent** [comment]
Hey, guys. I found an interesting workaround for using g++ 9.30 or newer versions on Ubuntu 20.04. 

Just revert the change in fix #432 . i.e. replace `-Wno-class-memaccess` with `-Wno-error=class-memaccess` in [premake5.lua file ](https://github.com/recastnavigation/recastnavigation/blob/master/RecastDemo/premake5.lua)

It seems that the change in #432 fixed bug for old gcc versions but raised new problems for new gcc versions. 

Help this info helps.

**Comment by grahamboree** [comment]
Fixed in #573.  Thanks @SgtVincent for the suggestion!
